app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,48 @@ def update
         end
       end
 
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @see https://github.com/zquestz/omniauth-google-oauth2#devise
+      #
+      def exist
```

Comment:
Reminder to delete all the code from this file that your ported over to the new controller